tests/testthat/test-estimate_and_apply_uncertainty.R

```diff
@@ -0,0 +1,97 @@
+triangle <- matrix(
+  data = sample.int(10, 12 * 5, replace = TRUE),
+  nrow = 12,
+  ncol = 5
+) |> construct_triangle()
+pt_nowcast_matrix <- estimate_and_apply_delay(
+  reporting_triangle = triangle,
+  n = 6
+)
+
+tv <- allocate_reference_times(triangle)
+n_delay_valid <- tv$n_history_delay
+n_retro_valid <- tv$n_retrospective_nowcasts
+
+
+test_that("estimate_and_apply_uncertainty works as expected with the default settings", { # nolint
+  set.seed(123)
+  nowcast_draws_df <- estimate_and_apply_uncertainty(
+    pt_nowcast_matrix,
+    triangle,
+    n_history_delay = n_delay_valid,
+    n_retrospective_nowcasts = n_retro_valid
+  )
+  set.seed(123)
+  df_w_default <- expect_message(
+    estimate_and_apply_uncertainty(
+      pt_nowcast_matrix,
+      triangle
+    ),
+    regexp = "Using 6 reference times for delay estimation." # nolint
+  )
+  expect_identical(nowcast_draws_df, df_w_default)
```

Comment:
Why are you testing it against itself like this?

Reply:
removed, replaced with tests against low level functions. Was just testing defaults and specification are same but weve moved to must specify